packages/vaadin-grid/src/vaadin-grid-keyboard-navigation-mixin.js

```diff
@@ -519,8 +519,11 @@ export const KeyboardNavigationMixin = (superClass) =>
         // Inform cell content of the focus (used in <vaadin-grid-sorter>)
         cell._content.dispatchEvent(new CustomEvent('cell-focusin', { bubbles: false }));
 
-        // Fire a public event for cell focus.
-        cell.dispatchEvent(new CustomEvent('cell-focus', { bubbles: true, composed: true }));
+        const context = this.getEventContext(e);
+        // Fire a public event for cell.
+        cell.dispatchEvent(
+          new CustomEvent('cell-focus', { bubbles: true, composed: true, detail: { context: context } })
```

Comment:
Why `event.detail.context.item` and not `event.detail.item` (where detail is the context)?

Reply:
Maybe some other info will be added in the `details` object? I don't know if that will ever happen, I think the pattern that we had for [GridCellActivateEvent](https://github.com/vaadin/web-components/blob/30f1f892f64a72242b53c4b78179a2763eb145a3/packages/vaadin-grid/src/vaadin-grid.d.ts#L49) is repeated here for `GridCellFocusEvent`,

Reply:
I guess because I suggested it like that in the related ticket. My reasoning for that suggestion was that there are other events where we could add the context at some point, and we would like to keep accessing the context between those events consistent. But these other events might already have other properties in the details object. So rather than mixing the context object with the other details, I suggested to put the context into its own property.